src/checkers/inference/solver/backend/maxsat/encoder/MaxSATImplicationConstraintEncoder.java

```diff
@@ -0,0 +1,88 @@
+package checkers.inference.solver.backend.maxsat.encoder;
+
+import checkers.inference.model.Constraint;
+import checkers.inference.model.ImplicationConstraint;
+import checkers.inference.solver.backend.encoder.implication.ImplicationConstraintEncoder;
+import checkers.inference.solver.backend.maxsat.MaxSatFormatTranslator;
+import checkers.inference.solver.frontend.Lattice;
+import checkers.inference.util.ConstraintVerifier;
+import org.sat4j.core.VecInt;
+
+import javax.lang.model.element.AnnotationMirror;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class MaxSATImplicationConstraintEncoder extends MaxSATAbstractConstraintEncoder implements ImplicationConstraintEncoder<VecInt[]> {
+
+    private final MaxSatFormatTranslator formatTranslator;
+
+    public MaxSATImplicationConstraintEncoder(Lattice lattice, ConstraintVerifier verifier,
+                                              Map<AnnotationMirror, Integer> typeToInt, MaxSatFormatTranslator formatTranslator) {
+        super(lattice, verifier, typeToInt);
+        this.formatTranslator = formatTranslator;
+    }
+
+    @Override
+    public VecInt[] encode(ImplicationConstraint constraint) {
```

Comment:
Can you add a high-level, human-readable description of what this code is doing?

Reply:
Sure. I'll add that